### ISSUE TITLE
Task.Delay 100x slower on mono than coreclr for macOS

### ISSUE THREAD

**Issue by trympet** [dotnet/runtime#57877]
### Description

On macOS, the overhead of `Task.Delay` is 20x higher on mono.

The behavior has been observed on:
2017 MacBook w/ Intel i5 x86
2020 Mac Mini w/ M1 arm64

Steps to reproduce:
1. `dotnet new macos` or create from VS template.
1. Open `AppDelegate.cs` and add the following:
```csharp
        public override void DidFinishLaunching(NSNotification notification)
        {
            // Insert code here to initialize your application
        _ = Task.Run(TaskBenchmark);
        _ = Task.Run(TaskBenchmark);
        _ = Task.Run(TaskBenchmark);
        _ = Task.Run(TaskBenchmark);
        }

        public static async Task TaskBenchmark()
        {
            int i = 0;
            while(i++ < 10000) {
                await Task.Delay(1).ConfigureAwait(false);
            }
        }
```
3. Open `top`, `Activity Monitor`, etc., and observe CPU usage of the process.

For reference, the coreclr equivalent is waay more performant.
1. Create a new .NET 5 console app from a template.
1. `dotnet new globaljson`; set SDK to 5.0.9.
1. Add the following and build with `dotnet build -c Release /p:UseAppHost=True`
```csharp
    public static void Main(string[] args)
    {
        _ = Task.Run(TaskBenchmark);
        _ = Task.Run(TaskBenchmark);
        _ = Task.Run(TaskBenchmark);
        _ = Task.Run(TaskBenchmark);
        Console.ReadKey();
    }

    public static async Task TaskBenchmark()
    {
        int i = 0;
        while(i++ < 10000) {
            await Task.Delay(1);
        }
    }
```

This completely breaks benign methods which rely on TPL for frequent polling.

### Configuration

OS=macOS Big Sur 11.5.2 (20G95) [Darwin 20.6.0]
Apple M1 2.40GHz, 1 CPU, 8 logical and 8 physical cores
Intel Core i5-7Y54 CPU 1.20GHz (Max: 1.30GHz) (Kaby Lake), 1 CPU, 4 logical and 2 physical cores
.NET SDK=5.0.400
### Data

Coreclr:
![image](https://user-images.githubusercontent.com/23460729/130319417-fc1935f8-2774-45cc-b47f-a2c26fc5ed8b.png)

Mono:
![image](https://user-images.githubusercontent.com/23460729/130319480-0dbcf0c2-2eae-4495-9043-52368c988c4c.png)

**Comment by EgorBo** [comment]
I'd guess it's not `Task.Delay` that is slow but the machinery around async/await (state machines) handling

**Comment by trympet** [comment]
> I'd guess it's not Task.Delay that is slow but the machinery around async/await (state machines) handling

Yes, sounds right. Using a plain old `System.Threading.Timer` provides acceptable perf.

However, the issue still remains; libraries which uses async await over explicit threading (e.g. render timers, async logging, etc.) can have widely different perf characteristics between Mono and coreclr. Having to fork every library which suffers because of this is less than ideal when multi-targeting.

**Comment by SamMonoRT** [comment]
@lambdageek @vargaz - marked for 7.0, but if there is a low risk fix, we can backport
